src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -1198,9 +1198,23 @@ def open_vm_port(cmd, resource_group_name, vm_name, port, priority=900, network_
 
     # update the NSG with the new rule to allow inbound traffic
     SecurityRule = cmd.get_models('SecurityRule', resource_type=ResourceType.MGMT_NETWORK)
-    rule_name = 'open-port-all' if port == '*' else 'open-port-{}'.format(port)
+
+    # use portranges unless wildcard is entered
+    if port == '*':
+        rule_name = 'open-port-all'
+        port_args = {
+            'destination_port_range': port,
+            'destination_port_ranges': None
+        }
+    else:
+        rule_name = 'open-port-{}'.format(port.replace(',', '_'))
+        port_args = {
+            'destination_port_range': None,
+            'destination_port_ranges': port.split(',')
```

Comment:
If `port` is a single port, `destination_port_range` seems better.

Reply:
changed so that unless a comma is present destination_port_range is used.
Examples:

555 -> destination_port_range
555-559 -> destination_port_range
\*  -> destination_port_range
555,556  -> destination_port_ranges